xml/array.go

```diff
@@ -0,0 +1,59 @@
+package xml
+
+// arrayMemberWrapper is the default member wrapper tag name for XML Array type
+const arrayMemberWrapper = "member"
+
+// Array represents the encoding of a XML array type
+type Array struct {
+	w       writer
+	scratch *[]byte
+
+	memberWrapperName  string
+	memberStartElement *StartElement
+	memberEndElement   *EndElement
```

Comment:
Yeah I guess, the pro here is it would alloc'd once and used throught the lifetime of this array. But stack allocations are cheaper, so if this can just flow downward it probably eliminates the need to keep on the heap and have it GC'd later.